### DISCUSSION TITLE
How can I save the inference results into json format?

### DISCUSSION THREAD

**Discussion by didnmj** [facebookresearch/detectron2#3989] [Q&A] [answered]
`from detectron2.evaluation import COCOEvaluator, inference_on_dataset
from detectron2.data import build_detection_test_loader
evaluator = COCOEvaluator("test", cfg, False, output_dir="./output2/")
val_loader = build_detection_test_loader(cfg, "test")
inference_on_dataset(trainer.model, val_loader, evaluator)`

**I think this code is for evaluation. I want to save the inference result for each image as json.**

**Comment by andreaceruti** [comment, marked as answer]
inference_on_dataset will save in your output directory what are you looking for, it should be the "coco_instances_result.json" or something similar.

**Reply by didnmj** [reply]
I want to save the predicted image in the coco format json.

**Reply by andreaceruti** [reply]
Have you seen that file in your output folder? It has all the result, you can visualize the annotations as the cocoAPI demo
[coco_api_notebook](https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocoDemo.ipynb)
